Classes/New Issue/IssueTemplates.swift

```diff
@@ -0,0 +1,220 @@
+                    title: template.title,
+                    style: .default,
+                    handler: { _ in
+
+                        guard let viewController = NewIssueTableViewController.createWithTemplate(
+                            client: GithubClient(userSession: session),
+                            owner: owner,
+                            repo: repo,
+                            template: template.template,
+                            signature: template.title == "Bug Report" ? .bugReport : .sentWithGitHawk
+                            ) else {
+                                Squawk.showGenericError()
+                                return
+                        }
+                        viewController.delegate = delegate
+                        let navController = UINavigationController(rootViewController: viewController)
+                        navController.modalPresentationStyle = .formSheet
+                        mainViewController.present(navController, animated: trueUnlessReduceMotionEnabled)
+                }))
+        }
+
+        alertView.addAction(
+            UIAlertAction(
+                title: "Dismiss",
+                style: UIAlertActionStyle.cancel,
+                handler: { _ in
+                    alertView.dismiss(animated: true, completion: nil)
+            })
+        )
+        mainViewController.present(alertView, animated: true, completion: nil)
+    }
+}
+
+extension GithubClient {
+
+    private func fetchTemplateFile(
+        owner: String,
+        repo: String,
+        filename: String,
+        completion: @escaping (Result<String>) -> Void
+        ) {
+
+        self.fetchFile(
+            owner: owner,
+            repo: repo,
+            branch: "master",
+            path: ".github/ISSUE_TEMPLATE/\(filename)") { (result) in
+                switch result {
+                case .success(let file):
+                    completion(.success(file))
+                case .error(let error):
+                    completion(.error(error))
+                case .nonUTF8:
+                    completion(.error(nil))
+                }
+        }
+    }
+
+    private func createTemplateWith(
+        owner: String,
+        repo: String,
+        filename: String,
+        completion: @escaping (Result<IssueTemplate>) -> Void
+        ) {
+        fetchTemplateFile(owner: owner, repo: repo, filename: filename) { (result) in
+            switch result {
+            case .success(let file):
+                let nameAndDescription = IssueTemplateHelper.getNameAndDescriptionFromTemplateFile(file: file)
+                if let name = nameAndDescription.name {
+                    completion(.success(IssueTemplate(title: name, template: file)))
+                } else {
+                    completion(.error(nil))
+                }
+            case .error(let error):
+                completion(.error(error))
+            }
+        }
+    }
+
+    func createNewIssue(
```

Comment:
Maybe split some of this up to make it more readable; it does a lot, making it hard to parse.